### ISSUE TITLE
Improve error messages

### ISSUE THREAD

**Issue by neingeist** [mrsimonemms/devpod-provider-hetzner#16]
### Bug description

When creating a workspace:
![image](https://github.com/mrsimonemms/devpod-provider-hetzner/assets/398881/e33bbba8-9235-49bc-99c5-d82810aa7df1)

I believe I had a somewhat generic error message ("(Server error)") before.

This turned out to be:

![image](https://github.com/mrsimonemms/devpod-provider-hetzner/assets/398881/a0c0e52c-8b1d-4cdb-bf59-11e9df5405a3)

**What irritates me is the "Usage" printout here - could this be improved?**

### Steps to reproduce

Hard to reproduce, because this seems to happen when Hetzner has issues

### Expected behaviour

Give a better error message.

**Comment by mrsimonemms** [comment]
Unless I'm reading it wrong, the error is already printed out (the SSH key uniqueness error).

The usage error is Cobra's standard setting of outputting the usage after all errors - I think it's [here in the docs](https://cobra.dev/#defining-your-own-usage).

I'm not against changing it, but it does seem like a lot of faff for not much benefit

**Comment by neingeist** [comment]
Yes but this was a second error! After removing the SSH key manually via Hetzner's Web interface (probably some inconsistent state after the Hetzner problems today), I get the original error:

![image](https://github.com/mrsimonemms/devpod-provider-hetzner/assets/398881/52d09aae-b78b-4e98-9f48-2637f522ecaf)

Any ideas on that?

**Comment by neingeist** [comment]
(Just for searchibility: The screenshot says

```
info Error:   (server_error)
```

**Comment by mrsimonemms** [comment]
Aaaah, now I understand the request - apologies.

It looks like it's a [Hetzner error code](https://docs.hetzner.cloud/#errors).

I think there is an improvement that can be made, which is to put the Hetzner error into a nicer format - instead of just `<err.code>`, something like `<err.code>: <err.message> - <err.details>`. It might get a bit big, but I think I can do something better than what it is now.

Thanks for the clarification.

**Comment by neingeist** [comment]
> Thanks for the clarification.

Pleased to clarify ;) I had it mixed-up a bit first, due to multiple problems at once (my mistake of changing the region while not thinking about the volume, Hetzner's hickup today + the inconsistent state of the workspace/leftover SSH key).

**Comment by neingeist** [comment]
> I think there is an improvement that can be made, which is to put the Hetzner error into a nicer format - instead of just `<err.code>`, something like `<err.code>: <err.message> - <err.details>`. It might get a bit big, but I think I can do something better than what it is now.

That does sound like it would be an improvment, at least I would expect that Hetzner API would tell me what is wrong. 

The actual problem might be easily reproducible by 1. changing the Hetzner region used in the provider and 2. rebuilding the workspace. I think this provider should then try to re-use the volume - and fail because of the region change. (I use DevPod and devpod-provider-hetzner since yesterday and might not understand the details properly.)

**Comment by neingeist** [comment]
A second thing I notice: The error was logged as `info`, should probably be at level `error`, if that's possible.

**Comment by mrsimonemms** [comment]
The error level is controlled by the DevPod application rather than the provider - everything it receives is on the `info` level.

It's loosely related to https://github.com/loft-sh/devpod/issues/1537
